components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self.containerView.backgroundColor = backgroundColor;
+}
+
+- (void)setTextFont:(UIFont *)textFont {
+  self.textLabel.font = textFont;
+}
+
+- (UIFont *)textFont {
+  return self.textLabel.font;
+}
+
+- (void)setTextColor:(UIColor *)textColor {
+  self.textLabel.textColor = textColor;
+}
+
+- (UIColor *)textColor {
+  return self.textLabel.textColor;
 }
 
-- (NSUInteger)numberOfButtons {
-  return self.buttons.count;
+- (void)setIconTintColor:(UIColor *)iconTintColor {
+  self.iconImageView.tintColor = iconTintColor;
+}
+
+- (UIColor *)iconTintColor {
+  return self.iconImageView.tintColor;
 }
 
 #pragma mark - UIView overrides
 
 - (CGSize)sizeThatFits:(CGSize)size {
-  self.layout = [[MDCBannerViewLayout alloc] initWithSizeToFit:size
-                                                     textLabel:self.textLabel
-                                                imageContainer:self.iconImageViewContainer
-                                                       buttons:self.buttons];
-  return [self.layout frameSize];
+  return CGSizeMake([UIScreen mainScreen].bounds.size.width, self.layout.size.height);
 }
 
-- (void)layoutSubviews {
-  [super layoutSubviews];
+- (CGSize)intrinsicContentSize {
+  return CGSizeMake([UIScreen mainScreen].bounds.size.width, self.layout.size.height);
+}
 
-  [self applyMDCBannerViewLayout:self.layout];
+- (void)updateConstraints {
+  if (!self.cachedLayout || self.cachedLayout != self.layout) {
+    [self updateConstraintsWithLayoutStyle:self.layout.style];
+    self.cachedLayout = self.layout;
+  }
+
+  [super updateConstraints];
 }
 
 #pragma mark - Layout methods
 
-- (void)applyMDCBannerViewLayout:(MDCBannerViewLayout *)layout {
-  if (![self isMDCBannerViewLayoutApplicable:layout]) {
-    return;
+- (void)setIsLayoutDirty:(BOOL)isLayoutDirty {
+  if (isLayoutDirty) {
+    [self invalidateIntrinsicContentSize];
   }
+  _isLayoutDirty = isLayoutDirty;
+}
 
-  self.iconImageViewContainer.frame = layout.imageContainerFrame;
-  self.iconImageView.center = CGPointMake(CGRectGetWidth(layout.imageContainerFrame) / 2,
-                                          CGRectGetHeight(layout.imageContainerFrame) / 2);
-  self.textLabel.frame = layout.textLabelFrame;
-  [self.buttons enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
-    if (idx >= layout.buttonFrames.count) {
-      *stop = YES;
-      return;
-    }
-    UIButton *button = (UIButton *)object;
-    CGRect buttonFrame = [layout.buttonFrames[idx] CGRectValue];
-    button.frame = buttonFrame;
-  }];
+- (MDCBannerViewLayout *)layout {
+  if (!_layout || self.isLayoutDirty) {
+    _layout = [[MDCBannerViewLayout alloc] initWithPreferredWidth:self.preferredContentWidth
+                                                        textLabel:self.textLabel
+                                                    iconContainer:self.iconImageView
+                                                          buttons:self.buttons];
+    self.isLayoutDirty = NO;
```

Comment:
I believe this can be removed.

Reply:
Same reason as mentioned above in https://github.com/material-components/material-components-ios/pull/6830#discussion_r264361551. It is to mark the flag clean after the layout recalculation.